### DISCUSSION TITLE
关于emqx与nanomq(Clustering between NanoMQ & EMQX)

### DISCUSSION THREAD

**Discussion by Pingchas** [nanomq/nanomq#960] [Q&A] [answered]
如何实现nanomq节点发送的消息，emqx节点能收到，并且emqx节点发送的消息，nanomq节点能收到。

**Comment by JaylinYu** [comment]
Hi, 可以通过桥接方式，把需要的主题从nanomq双向桥接到EMQX即可，但不能以集群的方式实现，

You can set up a bidirectional bridging connection on several topics with NanoMQ, however it is different from clustering.

**Comment by Pingchas** [comment]
我尝试了下双向桥接，但是发生了循环消息，是需要把主题消息挂到新的主题上吗？

**Comment by JaylinYu** [comment, marked as answer]
Hi ，如果你在nanomq上面同时对一个主题设置了forward/sub，那么就会产生回环消息。我理解你是可能需要类似集群的方式完成两个broker的消息互通，可以提出一个feature request。

If you enable both forwarding and sub on the same topic, it will cause a loopback msg, Apparently, you are demanding a clustering method to connect nanomq and emqx. This is not supported yet, but I hope we will put this on track if you raise a feature request.
